syft/internal/sourcemetadata/generated.go

```diff
@@ -2,9 +2,13 @@
 
 package sourcemetadata
 
-import "github.com/anchore/syft/syft/source"
+import (
+	"github.com/anchore/syft/syft/source/directorysource"
+	"github.com/anchore/syft/syft/source/filesource"
+	"github.com/anchore/syft/syft/source/stereoscopesource"
+)
 
 // AllTypes returns a list of all source metadata types that syft supports (that are represented in the source.Description.Metadata field).
 func AllTypes() []any {
-	return []any{source.DirectorySourceMetadata{}, source.FileSourceMetadata{}, source.StereoscopeImageSourceMetadata{}}
+	return []any{directorysource.Metadata{}, filesource.Metadata{}, stereoscopesource.ImageMetadata{}}
```

Comment:
though it feels clean to have `noun.Metadata` it ends up coupling implementation packages on the consumer side when type asserting the struct back out. Should we leave these as-is in `source` like we've done with `pkg`?